packages/patrol_cli/lib/src/features/test/test_command.dart

```diff
@@ -227,70 +235,80 @@ class TestCommand extends StagedCommand<TestCommandConfig> {
     config.devices.forEach(_testRunner.addDevice);
     _testRunner
       ..repeats = config.repeat
+      ..builder = (target, device) async {
+        Future<void> Function() action;
+
+        switch (device.targetPlatform) {
+          case TargetPlatform.android:
+            final options = AndroidAppOptions(
+              target: target,
+              flavor: config.flavor,
+              dartDefines: config.dartDefines,
+            );
+            action = () => _androidTestBackend.build(options, device);
```

Comment:
Instead of such switching, you could have a method that **returns** a callback

Reply:
It doesn't look that bad after I extracted _builder_ and _executor_ callbacks it to a private method. Gonna leave it as it is.